XMonad/Hooks/DynamicLog.hs

```diff
@@ -452,7 +452,7 @@ makeStatusBar lh sh k conf = pure $ docks $ conf
     { layoutHook  = avoidStruts (layoutHook conf)
     , logHook     = logHook conf *> lh
     , keys        = (<>) <$> keys' <*> keys conf
-    , startupHook = startupHook conf *> sh
+    , startupHook = cleanupStatusBars *> startupHook conf *> sh
```

Comment:
You probably want to add this to `statusBarPropTo` instead, because functions like `statusBar` do their own cleanup via the pipes and hence don't care about `cleanupStatusBars`

Reply:
Right, yes!